pkg/app/app_list_test.go

```diff
@@ -0,0 +1,249 @@
+
+- name: database
+  namespace: my-app
+  chart: bitnami/postgres
+  version: 11.6.22
+`,
+				"/path/to/helmfile.d/helmfile_3.yaml": `
+releases:
+- name: global
+  chart: incubator/raw
+  namespace: kube-system
+`,
+			}
+
+			app := appWithFs(&App{
+				OverrideHelmBinary:  DefaultHelmBinary,
+				glob:                filepath.Glob,
+				abs:                 filepath.Abs,
+				OverrideKubeContext: "default",
+				Env:                 tc.environment,
+				Logger:              logger,
+				valsRuntime:         valsRuntime,
+			}, files)
+
+			expectNoCallsToHelm(app)
+
+			if tc.ns != "" {
+				app.Namespace = tc.ns
+			}
+
+			if tc.selectors != nil {
+				app.Selectors = tc.selectors
+			}
+
+			var listErr error
+			out := testutil.CaptureStdout(func() {
+				listErr = app.ListReleases(configImpl{})
+			})
+
+			var gotErr string
+			if listErr != nil {
+				gotErr = listErr.Error()
+			}
+
+			if d := cmp.Diff(tc.error, gotErr); d != "" {
+				t.Fatalf("unexpected error: want (-), got (+): %s", d)
+			}
+
+			assert.Equal(t, tc.expected, out)
+		}()
+
+		testhelper.RequireLog(t, "app_list_test", bs)
+	}
+
+	t.Run("default environment includes all releases", func(t *testing.T) {
+		check(t, testcase{
+			environment: "default",
+			expected: `NAME                       	NAMESPACE  	ENABLED	INSTALLED	LABELS  	CHART           	VERSION
+logging                    	kube-system	true   	true     	        	incubator/raw   	       
+kubernetes-external-secrets	kube-system	true   	true     	        	incubator/raw   	       
+external-secrets           	default    	true   	true     	app:test	incubator/raw   	       
+my-release                 	default    	true   	true     	app:test	incubator/raw   	       
+disabled                   	kube-system	true   	false    	        	incubator/raw   	       
+test2                      	           	true   	true     	        	incubator/raw   	       
+test3                      	           	true   	true     	        	incubator/raw   	       
+cache                      	my-app     	true   	true     	app:test	bitnami/redis   	17.0.7 
+database                   	my-app     	true   	true     	        	bitnami/postgres	11.6.22
+global                     	kube-system	true   	true     	        	incubator/raw   	       
+`,
+		})
+	})
+
+	t.Run("fail on unknown environment", func(t *testing.T) {
+		check(t, testcase{
+			environment: "staging",
+			error:       `err: no releases found that matches specified selector() and environment(staging), in any helmfile`,
+		})
+	})
+
+	t.Run("list releases matching selector and environment", func(t *testing.T) {
+		check(t, testcase{
+			environment: "development",
+			selectors:   []string{"app=test"},
+			expected: `NAME            	NAMESPACE	ENABLED	INSTALLED	LABELS                                                    	CHART        	VERSION
+external-secrets	default  	true   	true     	app:test,chart:raw,name:external-secrets,namespace:default	incubator/raw	       
```

Comment:
It feels like there is another issue to be addressed here (separate to this work): i.e. the 'LABELS' includes the extra ones that are added by default _if_ selectors are defined, but not otherwise?

Reply:
Question for resolving the labels issue in the future is whether the default labels (chart etc) _should_ or _should not_ be included here consistently? i.e. should those extra labels be _excluded_ here, or should they be _included_ everywhere else?